README.md

```diff
@@ -1,58 +1,274 @@
+
+    Just run `lix download` and your dependencies will be perfectly in sync.
+
+- **Your entire `haxe_libraries` folder should be tracked in version control.**
+
+    NPM and Yarn have a package.json file. Yarn has a yarn.lock file. Hmm has a "hmm.json" file. With Lix, you just commit the whole "haxe_libraries" folder to version control, making it easy to track any changes to dependency versions.
+
+    This means whenever you switch branches, pull an update or merge changes it is easy to get the exact dependencies in sync.
+
+    It also means that if you are hacking away on a development version of a library, `git status` will remind you that the library you're editing is using a non-standard classpath, a hint that you should correct it before pushing your changes.
+
+- **Dependency versions should be scoped to each project, rather than global.**
+
+    Haxelib eventually supported this with `haxelib --newrepo` and a hidden `.haxelib/` folder.
+
+    With Lix and Haxe Shim, we look for the `haxe_libraries/` folder in the same place as the `.haxerc` file.
+
+    This is why you call `switchx scope create` when you introduce Lix to your project - it creates the `.haxerc` file that tells Haxe Shim to look here for the `haxe_libraries/` folder.
+
+## FAQ
+
+### What does this do differently to haxelib?
+
+Haxelib was built many years ago, before npm even existed. It was great at first, but suffers from a few limitations:
+
+- The only thing tracked in version control is your hxml files, which works when you install libraries directly from Haxelib, but is very hard to manage when installing dependencies from GitHub etc.
+    - This made it difficult to use when you had to use unreleased versions of a library, or a fork of a library.
+    - This also failed to track the exact versions of dependencies, meaning even if the right version of "tink_macros" was installed, the wrong version of "tink_core" might be installed, etc.
+- The way haxe and haxelib are tightly coupled makes it difficult to maintain, so development has really stalled for several years. (Haxe Shim breaks this dependency by instead using a standardised way to locate information about dependencies, making it much easier to build a tool like lix.)
+- Juraj, who started Lix, was actually the main maintainer of Haxelib for about two years, and resolved that replacing it was a more effective plan than trying to improve it.
+
+### How does this compare to hmm?
+
+Hmm is another tool that aims to improve package management for haxe projects. It does this by storing version information and installation instructions in "hmm.json", allowing you to restore state based on those instructions.
+
+It's a big step forward from Haxelib, but we think Lix is a step forward again.
+
+Reasons you might prefer hmm:
+
+- you do not want to install NodeJS or NPM.
+- you do not want to use Haxe Shim, you'd prefer to use normal Haxe and normal Haxelib.
+- ... possibly something to do with hxcpp?
```

Comment:
@back2dos @thomasuster this was a placholder after the conversation I saw you having on Gitter. Is Lix suitable for projects using hxcpp, hxjava, or hxcs where there are those weird compiler-dependent haxelibs?